src/bun.js/bindings/bindings.cpp

```diff
@@ -3467,7 +3467,7 @@ JSC::EncodedJSValue JSC__JSPromise__result(JSC::JSPromise* promise, JSC::VM* arg
     }
 }
 
-uint32_t JSC__JSPromise__status(const JSC::JSPromise* arg0, JSC::VM* arg1)
+[[ZIG_EXPORT(nothrow)]] uint32_t JSC__JSPromise__status(const JSC::JSPromise* arg0, JSC::VM* arg1)
 {
     switch (arg0->status(reinterpret_cast<JSC::VM&>(arg1))) {
```

Comment:
this reinterpret is invalid, it's only been working because `status()` does not use the VM. `arg1` is passed into `reinterpret_cast` as `JSC::VM*&` (a reference to a pointer) and then that gets reinterpreted as a reference to the VM itself -- i.e. we have gotten a `VM&` which references the 8 bytes where the pointer `arg1` is stored plus whatever happens to be after.

```
test.cpp:3:57: error: reference to stack memory associated with parameter 'vm' returned [-Werror,-Wreturn-stack-address]
    3 | VM &reinterpret(VM *vm) { return reinterpret_cast<VM &>(vm); }
      |                                                         ^~
```

```suggestion
    switch (arg0->status(*arg1)) {
```

fortunately we do not have any `reinterpret_cast<VM&>`, and we only have 4 other `reinterpret_cast<JSC::VM&>` which are all in this file and all used for promise methods that don't actually use the VM.